library/src/G4CMPEmpiricalNIEL.cc

```diff
@@ -0,0 +1,101 @@
+/***********************************************************************\
+ * This software is licensed under the terms of the GNU General Public *
+ * License version 3 or later. See G4CMP/LICENSE for the full license. *
+\***********************************************************************/
+
+/// \file library/include/G4CMPEmpiricalNIEL.hh
+/// \brief NIEL calculation with energy-dependent k. The function k(E) is
+/// taken from the Ionization yield measurement in Ge CDMSlite detector
+/// using photo-neutron sources.
+
+/// If energy dependent k is not used, NIEL calculation will be according 
+/// to Lindhard (Lewin Smith) yield with k value set by the user.
+///
+/// Paper DOI: https://doi.org/10.1103/PhysRevD.105.122002
+//
+// 20250212  David Sade
+
+#include "globals.hh"
+#include "G4CMPEmpiricalNIEL.hh"
+#include "G4ExceptionSeverity.hh"
+#include "G4CMPConfigManager.hh"
+#include "G4Material.hh"
+#include "G4Pow.hh"
+#include "G4SystemOfUnits.hh"
+#include <algorithm>
+#include <cmath>
+#include <cfloat>
+
+
+// return the fraction of the specified energy which will be deposited as NIEL
+// if an incoming particle is stopped in the specified material,
+// energy in native G4 energy units.
+//
+// The parameter k is a function of energy
+
+
+G4CMPEmpiricalNIEL::G4CMPEmpiricalNIEL() 
+    : Empklow(DBL_MIN), Empkhigh(DBL_MIN), 
+      EmpElow(DBL_MIN), EmpEhigh(DBL_MIN), 
+      EmpkFixed(DBL_MIN), EmpEDepK(true) {
+
+    G4CMPConfigManager* config = G4CMPConfigManager::Instance();
+
+    // Fetch values from G4CMPConfigManager or set defaults
+    Empklow = (config->GetEmpklow() != DBL_MIN) ? config->GetEmpklow() : 0.040;
+    Empkhigh = (config->GetEmpkhigh() != DBL_MIN) ? config->GetEmpkhigh() : 0.142;
+    EmpElow = (config->GetEmpElow() != DBL_MIN) ? config->GetEmpElow() : 0.39 * keV;
+    EmpEhigh = (config->GetEmpEhigh() != DBL_MAX) ? config->GetEmpEhigh() : 7.0 * keV;
+    EmpkFixed = (config->GetEmpkFixed() != DBL_MIN) ? config->GetEmpkFixed() : 0.158;
```

Comment:
Do you want to move the actual numerical defaults up to ConfigManager?  Then you don't need these conditionals at all, you just fetch the value from ConfigManager -- it's either the true default, or the user changed it.